lib/backend/etcd/etcd.go

```diff
@@ -92,21 +103,18 @@ func ConnectEtcdClient(config *api.ClientConfig) (DatastoreReadWriteInterface, e
 }
 
 // Create an entry in the datastore.  This errors if the entry already exists.
-func (c *EtcdClient) Create(d *DatastoreObject) (*DatastoreObject, error) {
-	// Special case the profile
-	switch t := d.Key.(type) {
-	case ProfileKey:
-		return t.Create(c, d)
+func (c *EtcdClient) Create(d *KVPair) (*KVPair, error) {
+	if co, ok := d.Key.(CreateOverrider); ok {
```

Comment:
Hmm, this was how i had originally thought of doing it, but I was worried that this would happen:

-  Create is called with a Profile which calls the override (because Profile key implements CreateOverrider)
-  Overridden create calls back into the client with a ProfilesTag ... but I think the ProfileTag also implements the CreateOverrider because the key has the profile key embedded in it.

Is that not the case?